src/main/webapp/js/richTextEditor.js

```diff
@@ -1,6 +1,49 @@
 /* eslint-disable camelcase */ // The property names are determined by external library (tinymce)
-var richTextEditorBuilder = {
-    getDefaultConfiguration: function() {
+function RichTextEditor(params) {
+
+    // ******************************************************************************************************
+    // Private properties
+    // ******************************************************************************************************
+
+    var MANDATORY_INIT_PARAMS = ['selector'];
+    var initParams = {};
+
+    var MANDATORY_ADDITIONAL_PARAMS = ['uploadImageId', 'uploadImageInputName', 'createImageUploadUrl'];
```

Comment:
I agree that there is some truth to that, but there is a limit on being "more abstract". Here, your "additionalParams" makes all image-related parameters compulsory. Is any part of the name "additionalParams" suggesting that? In fact, the name part "additional" itself already suggests that all its parameters are compulsory, i.e. additional (fittingly).

Which is why I suggested something like "imageUploadParams" earlier. Granted that it is not as generalize-able as "additionalParams", it serves its purpose more clearly.